### PR TITLE
Fix table extension filtering edge cases and batching efficiency

### PR DESCRIPTION
**Pull request by ThomasPe**
## Summary
- fix starts-with OData filter generation for empty prefixes
- escape single quotes in starts-with and count filters
- optimize transaction batching by materializing once and slicing in bounded chunks
- add null guards across extension entrypoints
- add regression test for empty-prefix starts-with queries

## Validation
- dotnet test src/Medienstudio.Azure.Data.Tables.Extensions.Tests/Medienstudio.Azure.Data.Tables.Extensions.Tests.csproj

### CODE REVIEW

File: src/Medienstudio.Azure.Data.Tables.Extensions.Tests/ExtensionTests.cs [hunk 1]
```diff
@@ -182,6 +182,15 @@ public async Task GetAllEntitiesStartingWithTest()
...
 
+    [TestMethod]
+    public async Task GetAllEntitiesStartingWithEmptyPrefixTest()
+    {
+        CreateTestData();
+
+        List<TableEntity> entities = await _tableClient.GetAllEntitiesStartingWithAsync<TableEntity>("PartitionKey", string.Empty);
+
+        Assert.AreEqual(3003, entities.Count);
+    }
```
[the remark this answers is not in the document]
Comment (ThomasPe) [reply src/Medienstudio.Azure.Data.Tables.Extensions.Tests/ExtensionTests.cs:193]: Fixed in fabe44f. Added a deterministic helper regression assertion for an apostrophe-ending prefix and storage-backed count coverage for an apostrophe-containing partition key.

File: src/Medienstudio.Azure.Data.Tables.Extensions/Extensions.cs [hunk 2]
```diff
@@ -198,7 +227,8 @@ public static async Task DeleteAllEntitiesByPartitionKeyAsync(this TableClient t
     /// <returns>The total number of rows in the table</returns>
     public static async Task<int> CountEntitiesAsync(this TableClient tableClient, string? partitionKey = null)
     {
-        string? filter = partitionKey is null ? null : TableClient.CreateQueryFilter($"PartitionKey eq {partitionKey}");
+        ArgumentNullException.ThrowIfNull(tableClient);
+        string? filter = partitionKey is null ? null : $"PartitionKey eq '{partitionKey.Replace("'", "''")}'";
```
[the remark this answers is not in the document]
Comment (ThomasPe) [reply src/Medienstudio.Azure.Data.Tables.Extensions/Extensions.cs]: Fixed in fabe44f. `CountEntitiesAsync` now uses `TableClient.CreateQueryFilter` so the SDK owns value formatting and escaping.

File: src/Medienstudio.Azure.Data.Tables.Extensions/Helpers.cs [hunk 3]
```diff
@@ -50,13 +50,23 @@ public static DateTimeOffset TicksKeyToDateTimeOffset(string ticksKey)
...
         {
-            return $"{column} ge '{prefix}'";
+            return $"{column} ge '{escapedPrefix}'";
         }
+
         char nextChar = (char)(lastChar + 1);
-        string prefixNext = prefix[..^1] + nextChar;
-        return $"{column} ge '{prefix}' and {column} lt '{prefixNext}'";
+        string prefixNext = escapedPrefix[..^1] + nextChar;
+        return $"{column} ge '{escapedPrefix}' and {column} lt '{prefixNext}'";
```
[the remark this answers is not in the document]
Comment (ThomasPe) [reply src/Medienstudio.Azure.Data.Tables.Extensions/Helpers.cs]: Fixed in fabe44f. The upper bound is now computed from the unescaped prefix, and both range bounds are escaped independently. Added a regression assertion for a prefix ending in an apostrophe.
